src/networking.c

```diff
@@ -2295,9 +2300,15 @@ int processMultibulkBuffer(client *c) {
         c->multibulklen = ll;
 
         /* Setup argv array on client structure */
-        if (c->argv) zfree(c->argv);
-        c->argv_len = min(c->multibulklen, 1024);
-        c->argv = zmalloc(sizeof(robj*)*c->argv_len);
+        if (c->multibulklen > CLINET_STATIC_ARGV_LEN) {
+            if (c->argv) zfree(c->argv);
```

Comment:
It's weird that c->argv is freed in here, I think this is no sense

Reply:
It should be moved outside `if else`.
@judeng Is that what you mean?

Reply:
I read all references of the `c->argv`,  the `zfree` in here is no needed. It would be more appropriate as an assert statement.

Reply:
@judeng agree with you, we should replace it with an assertion.
@scottlii let's do it as suggestion.

Reply:
ok